File: lincoln/ruby/lib/types/table/table_name.rb
class TableName
  include Comparable
  
  def initialize(scope, name)
    @scope = scope
    @name = name
  end
  
  attr_accessor :scope, :name

  def ==(o)
    return (o.class == TableName and o.scope == scope and o.name == name)
  end

  def hash
    return to_s.hash
  end
  
  def eql?(o)
    o.is_a? TableName && to_s == o.to_s
  end
  
  def to_s
    scope.to_s + "::" + name.to_s
  end
  
  def <=>(tn)
    return (to_s <=> tn.to_s)
  end
end

File: java/p2/src/p2/types/table/TableName.java
package p2.types.table;

import java.io.Serializable;
import java.util.Set;

import p2.types.function.TupleFunction;

public class TableName implements Comparable<TableName>, Serializable {
	
	public String scope;
	public String name;
	
	public TableName(String scope, String name) {
		this.scope = scope;
		this.name = name;
	}
	
	public String toString() {
		return scope + "::" + name;
	}
	
	public boolean equals(Object o) {
		return o instanceof TableName &&
				toString().equals(o.toString());
	}
	
	public int hashCode() {
		return toString().hashCode();
	}

	public int compareTo(TableName o) {
		return toString().compareTo(o.toString());
	}
	
}

